src/plugins/part.c

```diff
@@ -2726,172 +2022,34 @@ gboolean bd_part_set_part_type (const gchar *disk, const gchar *part, const gcha
     part_num--;
 
     cxt = get_device_context (disk, error);
-    if (!cxt) {
-        /* error is already populated */
-        bd_utils_report_finished (progress_id, (*error)->message);
-        return FALSE;
-    }
-
-    if (!set_part_type (cxt, part_num, type_guid, BD_PART_TABLE_GPT, error)) {
-        bd_utils_report_finished (progress_id, (*error)->message);
-        close_context (cxt);
+    if (!cxt)
         return FALSE;
-    }
 
-    if (!write_label (cxt, NULL, disk, FALSE, error)) {
-        bd_utils_report_finished (progress_id, (*error)->message);
+    ret = fdisk_get_partition (cxt, part_num, &pa);
```

Comment:
`pa` seems unused further in the code and besides an additional leak below, it might be worth to remove it,

Reply:
I didn't realize `fdisk_gpt_set_partition_attrs` doesn't actually need the partition context, fixed.